File: archive/Factorial.cs
using System;

namespace Factorial {
	public class Program {
		public static void Main(string[] args) {
			Console.WriteLine("Enter a number to find the factorial of: ");

			int number = Convert.ToInt32(Console.ReadLine());
			Console.WriteLine(Factorial(number));
		}
		public static long Factorial(int n) {
		  if (n < 2) {
		    return 1;
		  }
		  return n * Factorial(n - 1);
		}
	}
}

File: archive/factorial.pl
use strict;
use warnings;

sub factorial {
  my $n = shift;

  if ($n < 2) {
    return 1;
  }
  return $n * factorial($n - 1)
}

print "Enter a number to find the factorial of: ";
my $number;
$number = <>;
print factorial(int($number)), "\n";
